tests/monitoring/metrics.go

```diff
@@ -0,0 +1,139 @@
+			"kubevirt_vnc_active_connections":      true,
+			"kubevirt_console_active_connections":  true,
+
+			// virt-controller
+			// needs a migration - ignoring since already tested in - VM Monitoring, VM migration metrics
+			"kubevirt_vmi_migration_phase_transition_time_from_creation_seconds": true,
+			"kubevirt_vmi_migrations_in_pending_phase":                           true,
+			"kubevirt_vmi_migrations_in_scheduling_phase":                        true,
+			"kubevirt_vmi_migrations_in_running_phase":                           true,
+			"kubevirt_vmi_migration_succeeded":                                   true,
+			"kubevirt_vmi_migration_failed":                                      true,
+		}
+
+		It("should contain virt components metrics", func() {
+			metrics = fetchPrometheusMetrics(virtClient)
+
+			err = virtoperator.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = virtapi.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = virtcontroller.SetupMetrics(nil, nil, nil, nil, nil, nil, nil, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, metric := range operatormetrics.ListMetrics() {
+				if excludedMetrics[metric.GetOpts().Name] {
+					continue
+				}
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, nil)))
+			}
+		})
+
+		It("should have kubevirt_vmi_phase_transition_time_seconds buckets correctly configured", func() {
+			buckets := virtcontroller.PhaseTransitionTimeBuckets()
+
+			for _, bucket := range buckets {
+				labels := map[string]string{"le": strconv.FormatFloat(bucket, 'f', -1, 64)}
+
+				metric := operatormetrics.NewHistogram(
+					operatormetrics.MetricOpts{Name: "kubevirt_vmi_phase_transition_time_from_deletion_seconds"},
+					prometheus.HistogramOpts{},
+				)
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, labels)))
+			}
+		})
+	})
+})
+
+func fetchPrometheusMetrics(virtClient kubecli.KubevirtClient) *libmonitoring.QueryRequestResult {
+	metrics, err := libmonitoring.QueryRange(virtClient, "{__name__=~\"kubevirt_.*\"}", time.Now().Add(-1*time.Minute), time.Now(), 15*time.Second)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(metrics.Status).To(Equal("success"))
+	Expect(metrics.Data.ResultType).To(Equal("matrix"))
+	Expect(metrics.Data.Result).ToNot(BeEmpty(), "No metrics found")
+
+	return metrics
+}
+
+func setupVM(virtClient kubecli.KubevirtClient) {
+	vm := createRunningVM(virtClient)
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", 1)
+
+	By("Deleting the VirtualMachine")
+	err := virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", -1)
+}
+
+func createRunningVM(virtClient kubecli.KubevirtClient) *v1.VirtualMachine {
+	vmi := libvmifact.NewGuestless(libvmi.WithNamespace(testsuite.GetTestNamespace(nil)))
```

Comment:
we don't need to specify `libvmi.WithNamespace` since the the `Create` already has its own

Reply:
```
 [FAILED] Unexpected error:
      <*errors.errorString | 0xc002857c80>: 
      an empty namespace may not be set when a resource name is provided
      {
          s: "an empty namespace may not be set when a resource name is provided",
      }
  occurred
  In [BeforeEach] at: tests/libwait/wait.go:76 @ 03/26/24 11:04:46.857
```

Reply:
Oh yes, the libvmi.wait needs it.
Thank you!